test/integration/mobile-image-mounter-test.ts

```
@@ -0,0 +1,199 @@
+import { logger } from '@appium/support';
+import { expect } from 'chai';
+import { createHash } from 'crypto';
+import { promises as fs } from 'fs';
+import { after, before, describe } from 'mocha';
+
+import { Services } from '../../src/index.js';
+import type { MobileImageMounterServiceWithConnection } from '../../src/index.js';
+
+const log = logger.getLogger('MobileImageMounterService.test');
+log.level = 'debug';
+
+describe('MobileImageMounterService Integration', function () {
+  this.timeout(40000);
+
+  let serviceWithConnection: MobileImageMounterServiceWithConnection | null =
+    null;
+  const testUdid = process.env.UDID || '';
+
+  before(async function () {
+    if (!testUdid) {
+      this.skip();
+      return;
+    }
+
+    // Establish connection for all tests
+    try {
+      serviceWithConnection =
+        await Services.startMobileImageMounterService(testUdid);
+    } catch (error) {
+      this.skip();
```

let it fail. Dependant test will anyway be skipped if `before` section throws